lib/timers.js

```diff
@@ -580,13 +580,15 @@ function processImmediate() {
   while (immediate) {
     domain = immediate.domain;
 
-    if (!immediate._onImmediate)
-      continue;
```

Comment:
Not sure, but I think that would only protect against the very next immediate being cleared in the callback, and not anything else after the next one in the queue and we'd be back in an infinite loop? I'll test though.